solc/CommandLineInterface.cpp

```diff
@@ -1239,12 +1239,16 @@ bool CommandLineInterface::assemble(
 	map<string, yul::AssemblyStack> assemblyStacks;
 	for (auto const& src: m_sourceCodes)
 	{
-		auto& stack = assemblyStacks[src.first] = yul::AssemblyStack(m_evmVersion, _language);
+		auto& stack = assemblyStacks[src.first] = yul::AssemblyStack(
+			m_evmVersion,
+			_language,
+			_optimize ? OptimiserSettings::full() : OptimiserSettings::minimal()
```

Comment:
Why is it `full` here and `standard` in `Common.cpp`?

Reply:
Because this is assembly mode and `--optimize` here means to optimize including yul.